src/overlays/actors/ovl_Eff_Stk/z_eff_stk.c

```diff
@@ -28,16 +28,87 @@ ActorInit Eff_Stk_InitVars = {
     (ActorFunc)EffStk_Draw,
 };
 
-#endif
+void func_80BF0DE0(EffStk* arg0, PlayState* arg1);
 
-extern UNK_TYPE D_06009F60;
+void EffStk_Init(Actor* thisx, PlayState* play) {
+    EffStk* this = (EffStk*)thisx;
+    Actor_SetScale(&this->actor, 0.2f);
+    this->actionFunc = func_80BF0DE0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Init.s")
+void EffStk_Destroy(Actor* thisx, PlayState* play) {
+    EffStk* this = (EffStk*)thisx;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Destroy.s")
+void func_80BF0DE0(EffStk* arg0, PlayState* arg1) {
+    s16 temp_v0;
+    s16 temp_v0_2;
+    u16 temp_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/func_80BF0DE0.s")
+    if (Cutscene_IsCueInChannel(arg1, 0x200U) != 0) {
+        temp_v1 = arg1->csCtx.actorCues[Cutscene_GetCueChannel(arg1, 0x200U)]->id;
+        switch (temp_v1) {
+            case 2:
+                temp_v0_2 = arg0->unk146;
+                arg0->actor.draw = EffStk_Draw;
+                if (temp_v0_2 < 0x3C00) {
+                    arg0->unk146 = temp_v0_2 + 0x400;
+                    arg0->unk148 = Math_SinS(arg0->unk146) * -630.0f;
+                }
+                arg0->unk144 += 1;
+                break;
+            case 3:
+                temp_v0 = arg0->unk146;
+                arg0->actor.draw = EffStk_Draw;
+                if (temp_v0 < 0x3C00) {
+                    arg0->unk146 = temp_v0 + 0x400;
+                    arg0->unk148 = Math_SinS(arg0->unk146) * -630.0f;
+                }
+                arg0->unk144 -= 1;
+                break;
+            default:
+                arg0->actor.draw = NULL;
+                arg0->unk146 = 0;
+                arg0->unk148 = 0.0f;
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Update.s")
+void EffStk_Update(Actor* thisx, PlayState* play) {
+    EffStk* this = (EffStk*)thisx;
+    this->actionFunc(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Draw.s")
+extern s32 D_06008920;
+extern s32 D_06008A38;
```

Comment:
This symbols are segmented symbols (you can tell because they have a name like `D_06XXXXXX`)
Those come from the object associated to this actor. You can check it in the initvars from this actor.

For this actor is `object_stk2`. You should be able to find the symbols corresponding to this offset in the corresponding xml file for the object (the xmls for the objects are in `assets/xml/objects/`)